master/internal/db/postgres_trial_metrics.go

```diff
@@ -172,3 +316,35 @@ func GetMetrics(ctx context.Context, trialID, afterBatches, limit int,
 
 	return res, err
 }
+
+// shallowUnionMetrics unions non-overlapping AvgMetrics of two metrics bodies.
+func shallowUnionMetrics(oldBody, newBody *metricsBody) (*metricsBody, error) {
+	if oldBody == nil {
+		return newBody, nil
+	}
+
+	if newBody == nil {
+		return oldBody, nil
+	}
+
+	// FIXME: We don't need to convert this
+	oldAvgMetricsMap := oldBody.AvgMetrics.AsMap()
+	newAvgMetricsMap := newBody.AvgMetrics.AsMap()
+	for key, newValue := range newAvgMetricsMap {
+		// we cannot calculate min/max efficiently for replaced metric values
+		// so we disallow it.
+		if _, ok := oldAvgMetricsMap[key]; ok {
+			fmt.Println("old, new", oldBody.AvgMetrics, newBody.AvgMetrics)
```

Comment:
debug

Reply:
I didn't mean to keep it I just removed it but I can keep it as debug too